app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx

```diff
@@ -0,0 +1,129 @@
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { Responsive, WidthProvider } from 'react-grid-layout';
+import classNames from 'classnames/bind';
+import { activeProjectSelector } from 'controllers/user';
+import { fetch } from 'common/utils';
+import { ScrollWrapper } from 'components/main/scrollWrapper';
+import PropTypes from 'prop-types';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import styles from './widgetsGrid.scss';
+
+const cx = classNames.bind(styles);
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+@connect((state) => ({
+  url: `/api/v1/${activeProjectSelector(state)}/dashboard`,
+}))
+export class WidgetsGrid extends PureComponent {
+  static propTypes = {
+    url: PropTypes.string,
+    isFullscreen: PropTypes.bool,
+    dashboardId: PropTypes.string,
+  };
+
+  static defaultProps = {
+    url: '',
+    isFullscreen: false,
+    dashboardId: '',
+  };
+
+  state = {
+    widgets: [],
+    isFetching: false,
+  };
+
+  componentDidMount() {
+    this.fetchWidgets();
+  }
+
+  onBreakpointChange = (newBreakpoint) => {
+    this.isMobile = /sm|xs/.test(newBreakpoint);
+  };
+
+  onGridChange = (newLayout) => {
+    let newWidgets;
+
+    if (this.isMobile) {
+      const oldWidgets = this.state.widgets;
+
+      newWidgets = newLayout.map(({ i, y, h }, index) => ({
+        widgetId: i,
+        widgetPosition: [oldWidgets[index].widgetPosition[0], y],
+        widgetSize: [oldWidgets[index].widgetSize[0], h],
+      }));
+    } else {
+      newWidgets = newLayout.map(({ i, x, y, w, h }) => ({
+        widgetId: i,
+        widgetPosition: [x, y],
+        widgetSize: [w, h],
+      }));
+    }
+
+    this.setState({ widgets: newWidgets });
+    this.updateWidgets(newWidgets);
+  };
+
+  updateWidgets(widgets) {
+    const { dashboardId } = this.props;
+
+    fetch(`${this.props.url}/${dashboardId}`, {
+      method: 'PUT',
+      data: {
+        updateWidgets: widgets,
+      },
+    });
+  }
+
+  fetchWidgets() {
+    this.setState({ isFetching: true });
+
+    return fetch(this.props.url).then((result) => {
```

Comment:
I suggest fetching current dashboard by id and taking widgets from the response.

Reply:
Implemented